lib/util.js

```diff
@@ -460,3 +464,161 @@ module.exports = exports = {
                   'util.puts is deprecated. Use console.log instead.',
                   'DEP0027')
 };
+
+if (experimentalReport) {
+  try {
+    // If report is enabled, extract the binding and
+    // wrap the APIs with thin layers, with some error checks.
+    // user options can come in from CLI / ENV / API.
+    // CLI and ENV is intercepted in C++ and the API call here (JS).
+    // So sync up with both sides as appropriate - initially from
+    // C++ to JS and from JS to C++ whenever the API is called.
+    // Some events are controlled purely from JS (signal | exception)
+    // and some from C++ (fatalerror) so this sync-up is essential for
+    // correct behavior and alignment with the supplied tunables.
+    const nr = internalBinding('report');
```

Comment:
@joyeecheung - yes, it is to cover for possible errors from`internalBinding('report') in case node is built without `report` option.

`process.config.variables..` is throwing exception at the bootup for me:

```error
util.js:474
  if (process.config.variables.node_report === true) {
                     ^

TypeError: Cannot read property 'variables' of undefined
    at util.js:474:22
    at NativeModule.compile (internal/bootstrap/loaders.js:342:5)
    at Function.NativeModule.require (internal/bootstrap/loaders.js:213:16)
    at setupGlobalVariables (internal/bootstrap/node.js:432:29)
    at startup (internal/bootstrap/node.js:43:3)
    at internal/bootstrap/node.js:732:1
```

`nternalBinding('config').hasReport` approach: I will try.

Reply:
@joyeecheung - I took your firrst approach - `internalBinding('config').hasReport` and eliminated the need of a try-catch block thanks.